### ISSUE TITLE
Add a mechanism to "delete" published dandisets

### ISSUE THREAD

**Issue by mvandenburgh** [dandi/dandi-archive#1123]
This issue will require some heavy planning, including guidelines for policies/procedures around deleting published data (i.e., how dandiset owners should flag this situation for admin attention, whether they should be able to hide (and for how long) the dandiset from the public on their own authority, questions about scrubbing data permanently from the S3 bucket, etc.), and UI features to enable any/all of this.

- [ ] write a design doc

**Comment by mvandenburgh** [comment]
@satra when deleting a published dandiset, do we have to actually delete the data from our database/s3, or is simply hiding it from public view sufficient?

**Comment by bendichter** [comment]
Yes, as we are a free data storage platform, we should also prepare for the case where a user uploads illicit material and tries to use DANDI to distribute it
